test/core/xds/xds_client_test.cc

```diff
@@ -4128,28 +4113,33 @@ TEST_F(XdsClientTest, FallbackOnStartup) {
           .set_nonce("A")
           .AddFooResource(XdsFooResource("foo1", 6))
           .Serialize());
+  EXPECT_THAT(GetServerConnections(),
+              ::testing::ElementsAre(
+                  ::testing::Pair(kDefaultXdsServerUrl, false),
+                  ::testing::Pair(fallback_server.server_uri(), true)));
   // XdsClient should have delivered the response to the watcher.
   auto resource = watcher->WaitForNextResource();
   ASSERT_NE(resource, nullptr);
   EXPECT_EQ(resource->name, "foo1");
   EXPECT_EQ(resource->value, 6);
+  // Client sends an ACK.
   request = WaitForRequest(fallback_stream.get());
   ASSERT_TRUE(request.has_value());
   CheckRequest(*request, XdsFooResourceType::Get()->type_url(),
                /*version_info=*/"1", /*response_nonce=*/"A",
                /*error_detail=*/absl::OkStatus(),
                /*resource_names=*/{"foo1"});
   // Recover to primary
-  primary_stream = WaitForAdsStream(primary_server);
   primary_stream->SendMessageToClient(
       ResponseBuilder(XdsFooResourceType::Get()->type_url())
           .set_version_info("5")
           .set_nonce("D")
           .AddFooResource(XdsFooResource("foo1", 42))
           .Serialize());
-  ASSERT_NE(primary_stream, nullptr);
-  fallback_stream.reset();
+  EXPECT_TRUE(fallback_stream->Orphaned());
   resource = watcher->WaitForNextResource();
+  EXPECT_THAT(GetServerConnections(), ::testing::ElementsAre(::testing::Pair(
```

Comment:
Please move this down to line 4146, so that it's not right in between getting the resource from the watcher and checking the resource's value.

Reply:
Done. Thanks!